Source/ButtonContext.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package cs505.group1.state;
+
+/**
+ * @author CS505 Fall17 Group1
+ * @version 2017.10.29
+ */
+abstract class ButtonContext {
+    
+    private ButtonState buttonState;
+    
+    /**
+     * Constructor.  In concrete extensions buttonState  
+     * must be assigned an initial value such as
+     * this.buttonState =  new StateA();
+     */
+    public ButtonContext(){
+        this.buttonState = null;
+    }
+    
+    /**
+     * Implements code for a single button press.
+     * @return State allows a state change 
+     */
+    public abstract ButtonState singlePress();
```

Comment:
Having the singlePress return ButtonState on the actual state works well for communicating what the next state should be, but it isn't appropriate on the context.  On the context you essentially want to make outside classes unaware of these internal transitions.  So when I call singlePress on the context , have it either return void or something meaningful.  Internally the State can return what the next State is without exposing in externally.

Reply:
Changed as suggested.